x/gamm/types/msgs.go

```diff
@@ -18,6 +22,35 @@ const (
 	TypeMsgExitSwapShareAmountIn   = "exit_swap_share_amount_in"
 )
 
+func ValidateFutureOwner(owner string) error {
+	// allow empty governer
+	if owner == "" {
+		return nil
+	}
+	// validation for future owner
+	_, err := sdk.AccAddressFromBech32(owner)
+	if err == nil {
+		return nil
+	}
+
+	splits := strings.Split(owner, ",")
+	if len(splits) != 2 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid future owner: %s", owner))
+	}
+
+	lpTokenStr := splits[0]
+	if sdk.ValidateDenom(lpTokenStr) != nil {
```

Comment:
Will we know the denom of the LP token for a pool before creating the pool?

Reply:
Then we provide only information of own pool LP token?
Predetermined string like "PoolLPToken,1w" or remove it as well, just do "1w".
Is there any case, another token could be governance token of the pool?

Reply:
I think its fine to allow `this_pool_lp_token` to be a special word thats interpreted as this pool's LP token, and otherwise have any other string placed there be interpreted as just a token denom.

I think there will be cases where another token can be the governance token of the pool.

Reply:
Yeah, I think either of these two will be fine.  Just a heads up, the duration should be allowed to be 0w as well

> "PoolLPToken,1w" or remove it as well, just do "1w"

Reply:
Ok

Reply:
It's done